apex/optimizers/fused_adam.py

```diff
@@ -230,6 +140,15 @@ def step(self, closure=None, grads=None, output_params=None, scale=None, grad_no
         if closure is not None:
             loss = closure()
 
+        # The full precision params are set up in the first step of the optimizer
+        # instead of in the constructor because the full precision params will get out
+        # out of sync with the model params if DDP syncs the model params across devices
+        # after the optimizer is constructed.
+        # Also with CUDA Graphs, setting optimizer.param_groups_master = [] tells the
+        # optimizer to reconstruct master weights, which may be needed after graph capture.
+        if len(self.param_groups_master) == 0:
+            self._setup_master_params()
```

Comment:
guessing this needs to be guarded with `master_weights` value
```suggestion
        if self.master_weights and len(self.param_groups_master) == 0:
            self._setup_master_params()
```